### PR TITLE
Meets #14433: Long version names with long wiki name destroys layout

### PR DESCRIPTION
**Pull request by slawa85**
[`* `#14433` Long version names with long wiki name destroys layout`](https://www.openproject.org/work_packages/14433)

### CODE REVIEW

File: app/views/versions/show.html.erb [hunk 1]
```diff
@@ -29,11 +29,16 @@ See doc/COPYRIGHT.rdoc for more details.
 
 <% content_for :action_menu_specific do %>
   <%= link_to_if_authorized l(:button_edit), {:controller => '/versions', :action => 'edit', :id => @version}, :class => 'icon icon-edit' %>
-  <%= link_to_if_authorized(l(:button_edit_associated_wikipage, :page_title => @version.wiki_page_title), {:controller => '/wiki', :action => 'edit', :project_id => @version.project, :id => Wiki.titleize(@version.wiki_page_title)}, :class => 'icon icon-edit') unless @version.wiki_page_title.blank? || @version.project.wiki.nil? %>
+  <%= link_to_if_authorized(l(:button_edit_associated_wikipage, :page_title => truncate(@version.wiki_page_title, :length => 50, :separator => ' ')),
+                            {:controller => '/wiki', :action => 'edit',
+                             :project_id => @version.project,
+                             :id => Wiki.titleize(@version.wiki_page_title)},
+                            :title => Wiki.titleize(@version.wiki_page_title),
+                            :class => 'icon icon-edit tooltip') unless @version.wiki_page_title.blank? || @version.project.wiki.nil? %>
```
Comment (hschink) [inline app/views/versions/show.html.erb:37]: :+1:

File: app/views/versions/show.html.erb [hunk 2]
```diff
@@ -29,11 +29,16 @@ See doc/COPYRIGHT.rdoc for more details.
...
   <%= call_hook(:view_versions_show_contextual, { :version => @version, :project => @project }) %>
 <% end %>
 
-<h2><%= h(@version.name) %></h2>
+<h2><%= h(truncate(@version.name, :length => 50, :separator => ' ')) %></h2>
```
Comment (hschink) [inline app/views/versions/show.html.erb]: Would it be possible to set the `title` of `h2` and to use `text-overflow: ellipsis` instead?

Comment (slawa85) [inline app/views/versions/show.html.erb]: _h2_ tag didn't accept _title_ attribute

Comment (cratz) [inline app/views/versions/show.html.erb]: Hello,

title is a global HTML attribute and can be used on any HTML element see:
http://www.w3schools.com/tags/att_global_title.asp

Comment (slawa85) [inline app/views/versions/show.html.erb]: Done

File: app/views/versions/show.html.erb [hunk 3]
```diff
@@ -29,11 +29,16 @@ See doc/COPYRIGHT.rdoc for more details.
...
+                            {:controller => '/wiki', :action => 'edit',
+                             :project_id => @version.project,
+                             :id => Wiki.titleize(@version.wiki_page_title)},
+                            :title => Wiki.titleize(@version.wiki_page_title),
+                            :class => 'icon icon-edit tooltip') unless @version.wiki_page_title.blank? || @version.project.wiki.nil? %>
   <%= call_hook(:view_versions_show_contextual, { :version => @version, :project => @project }) %>
 <% end %>
 
-<h2><%= h(@version.name) %></h2>
+<h2 title="<%= h(@version.name)%>" class="ellipsis"><%= h(@version.name) %></h2>
```
Comment (hschink) [inline app/views/versions/show.html.erb]: This is actually not truncated. I guess a width is missing for ellipsis to work properly. You may have a look at different approaches, for instance [flexible box layout](http://stackoverflow.com/a/12656699) to fix this.

File: app/assets/stylesheets/default/main.css.erb [hunk 4]
```diff
@@ -303,6 +303,13 @@ div#roadmap .wiki h2 { font-size: 110%; }
 div#version-summary { float: right; width: 380px; margin-left: 16px; margin-bottom: 16px; background-color: #fff; }
 div#version-summary fieldset { margin-bottom: 1em; }
 div#version-summary .total-hours { text-align: right; }
+.page-title{
+    display: block;
+    width: 30%;
```
Comment (hschink) [inline app/assets/stylesheets/default/main.css.erb]: This approach does only take a fraction of the available space into account. We need a different approach here that adapts to the complete available width.

**Comment by hschink** [comment]
Works as advertised and can be merged as soon as Travis is done.

**Pull request merged**